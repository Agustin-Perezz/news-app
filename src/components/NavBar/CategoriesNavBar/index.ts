export * from './CategoriesNavBar';
export * from './listCategories';
