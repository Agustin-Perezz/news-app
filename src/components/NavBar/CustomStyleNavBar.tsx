import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';

export const CustomBoxContainer = styled(MuiBox)(() => ({
  backgroundColor: '#292929',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: 'max-content',
  flexWrap: 'wrap',
  paddingTop: '20px',
}));

export const CustomNavBarBox = styled(MuiBox)(() => ({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  width: '600px',
  margin: '0px 0px 20px 0px',
  '@media (min-width: 1003px)': {
    width: '880px',
    justifyContent: 'space-around',
  },
}));
