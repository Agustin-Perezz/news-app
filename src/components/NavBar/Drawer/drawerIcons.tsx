import AllOutIcon from '@mui/icons-material/AllOut';
import HomeIcon from '@mui/icons-material/Home';
import DashboardIcon from '@mui/icons-material/Dashboard';
import ScienceIcon from '@mui/icons-material/Science';
import SportsFootballIcon from '@mui/icons-material/SportsFootball';
import PaidIcon from '@mui/icons-material/Paid';
import LocalHospitalIcon from '@mui/icons-material/LocalHospital';
import SportsEsportsIcon from '@mui/icons-material/SportsEsports';
import DeveloperBoardIcon from '@mui/icons-material/DeveloperBoard';
import AutoStoriesIcon from '@mui/icons-material/AutoStories';
import IcecreamIcon from '@mui/icons-material/Icecream';
import ConnectingAirportsIcon from '@mui/icons-material/ConnectingAirports';

export const drawerIcons = [
  { icon: <AllOutIcon /> },
  { icon: <HomeIcon /> },
  { icon: <DashboardIcon /> },
  { icon: <ScienceIcon /> },
  { icon: <SportsFootballIcon /> },
  { icon: <PaidIcon /> },
  { icon: <LocalHospitalIcon /> },
  { icon: <SportsEsportsIcon /> },
  { icon: <DeveloperBoardIcon /> },
  { icon: <AutoStoriesIcon /> },
  { icon: <IcecreamIcon /> },
  { icon: <ConnectingAirportsIcon /> },
];
