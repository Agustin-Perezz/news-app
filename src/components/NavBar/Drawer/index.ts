export * from './DrawerCategories';
export * from './DrawerCategory';
export * from './NavBarDrawer';
export * from './drawerIcons';
