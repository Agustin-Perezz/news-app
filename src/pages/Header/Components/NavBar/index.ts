export * from './NavBar';
export * from './CustomStylesNavBar';
