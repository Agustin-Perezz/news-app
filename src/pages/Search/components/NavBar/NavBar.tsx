import { useContext } from 'react';
import { ImageListItem, Typography } from '@mui/material';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { SearchBar } from '../SearchBar';
import { NewsContext } from '../../../../context/NewsContext';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  return (
    <CustomStack
      stackProps={{
        direction: 'row',
        justifyContent: 'space-evenly',
        alignItems: 'center',
        marginBottom: -1.5,
      }}
    >
      <ImageListItem sx={{ width: 190 }}>
        <img src={news_logo} alt="news-logo" />
      </ImageListItem>
      <SearchBar />
      <SocialMedia />
    </CustomStack>
  );
};
