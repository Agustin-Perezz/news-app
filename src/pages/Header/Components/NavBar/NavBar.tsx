import Box from '@mui/material/Box';
import { ImageListItem } from '@mui/material';
import { SearchBar } from '../SearchBar';
import { CategoriesNavBar } from '../CategoriesNavBar';
import { CustomStack } from '../../../../components/CustomStack';
import { SocialMedia } from '../../../../components/ui/SocialMedia';
import { useSearch } from '../../hooks';

import news_logo from '../../../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  return (
    <Box
      sx={{
        backgroundColor: '#292929',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: 'max-content',
        flexWrap: 'wrap',
        paddingTop: '20px',
      }}
    >
      <CustomStack
        stackProps={{
          direction: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          width: '1000px',
        }}
      >
        <ImageListItem sx={{ width: 150 }}>
          <img src={news_logo} alt="news-logo" />
        </ImageListItem>
        <SearchBar handleSubmit={handleSubmit} />
        <SocialMedia />
      </CustomStack>
      <CategoriesNavBar handleClick={handleSubmit} />
    </Box>
  );
};
