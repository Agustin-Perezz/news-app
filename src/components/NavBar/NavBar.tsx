import { ImageListItem } from '@mui/material';
import useMediaQuery from '@mui/material/useMediaQuery';
import { SearchBar } from './SearchBar';
import { SocialMedia } from '../ui/SocialMedia';
import { CategoriesNavBar } from './CategoriesNavBar';
import { NavBarDrawer } from './Drawer';
import { CustomBoxContainer, CustomNavBarBox } from './CustomStyleNavBar';
import { CategoryProvider } from '../../context/CategoryContext';
import { useSearch } from '../../hooks';

import news_logo from '../../assets/images/news-logo.png';

export const NavBar: React.FC = () => {
  const { handleSubmit } = useSearch();
  const matches = useMediaQuery('(max-width:1000px)');

  return (
    <CategoryProvider>
      <CustomBoxContainer>
        <CustomNavBarBox>
          {matches ? (
            <NavBarDrawer handleSubmit={handleSubmit} />
          ) : (
            <ImageListItem sx={{ maxWidth: '130px' }}>
              <img src={news_logo} alt="news-logo" />
            </ImageListItem>
          )}
          <SearchBar handleSubmit={handleSubmit} />
          {!matches && <SocialMedia />}
        </CustomNavBarBox>
        {!matches && <CategoriesNavBar handleClick={handleSubmit} />}
      </CustomBoxContainer>
    </CategoryProvider>
  );
};
