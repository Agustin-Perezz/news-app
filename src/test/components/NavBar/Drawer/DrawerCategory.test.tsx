import { fireEvent, render, screen } from '@testing-library/react';
import { DrawerCategory } from '../../../../components/NavBar/Drawer';

type Props = { activeCategory: number; idxCategory: number };
describe('Test in <DrawerCategory />', () => {
  const handleSetTab = jest.fn();
  function setup({ activeCategory, idxCategory }: Props) {
    return render(
      <DrawerCategory
        activeCategory={activeCategory}
        icon={<span>TestIcon</span>}
        idxCategory={idxCategory}
        text="General"
        handleSetTab={handleSetTab}
      />
    );
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { container } = setup({ activeCategory: 0, idxCategory: 1 });
    expect(container).toMatchSnapshot();
  });
  it('should show correctly props', () => {
    setup({ activeCategory: 0, idxCategory: 1 });
    screen.getByText('General');
    screen.getByText('TestIcon');
  });
  it('should add active class when activeCategory and idxCategory are equal', () => {
    const { container } = setup({ activeCategory: 1, idxCategory: 1 });
    const itemCategory = container.firstChild;
    expect(itemCategory).toHaveClass('Mui-selected');
  });
  it('should call handleSetTab', () => {
    const { container } = setup({ activeCategory: 0, idxCategory: 1 });
    fireEvent.click(container.firstChild!);
    expect(handleSetTab).toHaveBeenCalled();
  });
});
