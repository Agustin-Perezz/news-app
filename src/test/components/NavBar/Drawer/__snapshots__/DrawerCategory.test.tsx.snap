// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <DrawerCategory /> should match snapshot 1`] = `
<div>
  <div
    class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1ahq8ay-MuiButtonBase-root-MuiListItemButton-root"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root css-wg4rhx-MuiListItemIcon-root"
    >
      <span>
        TestIcon
      </span>
    </div>
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        General
      </span>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </div>
</div>
`;
